src/anndata/_core/xarray.py

```diff
@@ -1,86 +1,151 @@
 from __future__ import annotations
 
 import warnings
+from collections.abc import Hashable, Mapping
+from typing import TYPE_CHECKING, overload
 
+import numpy as np
 import pandas as pd
 
 from ..compat import XDataArray, XDataset, XVariable
 
+if TYPE_CHECKING:
+    from collections.abc import Iterable, Iterator
+    from typing import Any, Literal
 
-class Dataset2D(XDataset):
+
+# See https://github.com/pydata/xarray/blob/main/xarray/core/dataset.py#L194 for typing
+class Dataset2D(Mapping[Hashable, "XDataArray | Dataset2D"]):
     """
-    A wrapper class meant to enable working with lazy dataframe data.
-    We do not guarantee the stability of this API beyond that guaranteed
-    by :class:`xarray.Dataset` and the `to_memory` function, a thin wrapper
-    around :meth:`xarray.Dataset.to_dataframe` to ensure roundtrip
-    compatibility here.
+
+    Bases :class:`~collections.abc.Mapping` [ :class:`~collections.abc.Hashable`, :class:`~xarray.DataArray` | :class:`~anndata.experimental.backed.Dataset2D` ]
```

Comment:
this should work

```suggestion
    r"""
    Bases :class:`~collections.abc.Mapping`\ [:class:`~collections.abc.Hashable`, :class:`~xarray.DataArray` | :class:`~anndata.experimental.backed.Dataset2D`\ ]
```

Reply:
Ah is it to escape the `\ `? Not sure I am familiar with the mechanism here

Reply:
Huh also the `r"""` was added in, I think by the formatter.  I don't remember doing it so still curious what is going on here, I thought that a raw string would not escape the `\` but must be misunderstanding